src/app/core/export/download-service/download.service.ts

```diff
@@ -90,17 +91,59 @@ export class DownloadService {
    * @returns string a valid CSV string of the input data
    */
   async createCsv(data: any[]): Promise<string> {
-    // Collect all properties because papa only uses the properties of the first object
+    let entityConstructor: any;
+
+    if (data.length > 0 && typeof data[0]?.getConstructor === 'function') {
+        entityConstructor = data[0].getConstructor();
+    }
     const keys = new Set<string>();
     data.forEach((row) => Object.keys(row).forEach((key) => keys.add(key)));
 
     data = data.map(transformToReadableFormat);
+   
+    if (!entityConstructor) {
+      return this.papa.unparse(data, {
+        quotes: true,
+        header: true,
+        newline: DownloadService.SEPARATOR_ROW,
+        columns: [...keys],
+      });
+    }
+
+    const result = this.exportFile(data, entityConstructor);
+    return result;
+  }
+
+  exportFile(data: any[], entityConstructor: { schema: any; }) {
+    const entitySchema = entityConstructor.schema;
+    const columnLabel = new Map<string, EntitySchemaField>();
+    
+    entitySchema.forEach((value: { label: string }, key: string) => {
+      if (value.label) columnLabel[key] = value.label;
+    });
+  
+    const exportEntities = data.map((item) => {
+      for (const key in item) {
+        if (!columnLabel.hasOwnProperty(key)) delete item[key];
+      }
+      return item;
+    });
+
+    const columnKeys = Object.keys(columnLabel);
+    const columnLabels:string[] = Object.values(columnLabel);
+    const orderedData = [];
+
+    exportEntities.forEach((item) => {
+      const orderedItem = [];
+      columnKeys.forEach((key) => {
+        orderedItem.push(item[key]);
+      });
+      orderedData.push(orderedItem);
+    });
 
-    return this.papa.unparse(data, {
-      quotes: true,
-      header: true,
-      newline: DownloadService.SEPARATOR_ROW,
```

Comment:
why no config for `newline` and `quotes` anymore?

Reply:
@sleidig I am Specifying fields and data explicitly, So I am not using for `newline` and `quotes`

Reply:
ah, I see your point. However, quotes and newline are config to specify how the output / result looks and we should still add them additionally.

I am pushing a small fix for this.